### ISSUE TITLE
sudo make install fails on Mint 17.2

### ISSUE THREAD

**Issue by quorth0n** [RicinApp/Ricin#120]
```
$ sudo make install

./waf distclean
'distclean' finished successfully (0.005s)
./waf configure
Setting top to                           : /home/liam/Ricin 
Setting out to                           : /home/liam/Ricin/build 
Checking for 'gcc' (C compiler)          : /usr/bin/gcc 
Checking for program 'pkg-config'        : /usr/bin/pkg-config 
Checking for 'gobject-2.0'               : yes 
Checking for 'gthread-2.0'               : yes 
Checking for program 'valac-0.8'         : not found 
Checking for program 'valac'             : /usr/bin/valac 
Checking for valac version >= (0, 8, 0)  : (0, 32, 0) 
Checking for program 'msgfmt'            : /usr/bin/msgfmt 
Checking for program 'perl'              : /usr/bin/perl 
Checking for program 'intltool-merge'    : /usr/bin/intltool-merge 
Checking for header locale.h             : yes 
Checking for program 'glib-genmarshal'   : /usr/bin/glib-genmarshal 
Checking for program 'glib-mkenums'      : /usr/bin/glib-mkenums 
Checking for program 'glib-compile-schemas' : /usr/bin/glib-compile-schemas 
Checking for program 'glib-compile-resources' : /usr/bin/glib-compile-resources 
Checking for program 'valac-0.28'             : /usr/bin/valac 
Checking for valac-0.28 version >= (0, 28, 0) : (0, 32, 0) 
Checking for 'glib-2.0'                       : yes 
Checking for 'gio-2.0'                        : yes 
Checking for 'gobject-2.0'                    : yes 
Checking for 'gmodule-2.0'                    : yes 
Checking for 'gtk+-3.0'                       : yes 
Checking for 'libsoup-2.4'                    : yes 
Checking for 'json-glib-1.0'                  : yes 
Checking for 'libnotify'                      : yes 
Checking for 'libtoxcore'                     : yes 
'configure' finished successfully (0.184s)
./waf build
Waf: Entering directory `/home/liam/Ricin/build'
[ 2/39] Compiling po/da_DK.po
[ 2/39] Compiling po/de.po
[ 3/39] Compiling po/en_US.po
[ 4/39] Compiling po/eo.po
[ 5/39] Compiling po/fr_FR.po
[ 6/39] Compiling po/pt_PT.po
[ 7/39] Compiling po/ru_RU.po
[ 8/39] Compiling po/uk.po
[ 9/39] Compiling po/zh_CN.po
[10/39] Compiling res/ricin.desktop.in
[11/39] Processing res/ricin.gresource.xml
print() on closed filehandle CACHE at /usr/bin/intltool-merge line 375, <PO_FILE> line 4913.

[12/39] Processing valac: res/ricin.gresource.xml src/ChatView.vala src/EditableLabel.vala src/FriendListRow.vala src/GroupListRow.vala src/HistoryManager.vala src/IMessageListRow.vala src/InlineFileMessageListRow.vala src/InlineImageMessageListRow.vala src/MainWindow.vala src/MessageListRow.vala src/Notification.vala src/PlainLabel.vala src/ProfileChooser.vala src/QuoteLabel.vala src/QuoteMessageListRow.vala src/Ricin.vala src/Settings.vala src/SettingsView.vala src/StatusMessageListRow.vala src/SystemMessageListRow.vala src/ThemeManager.vala src/ToxIdenticons.vala src/Util.vala src/WelcomeView.vala src/Wrapper.vala -> build/ChatView.c build/EditableLabel.c build/FriendListRow.c build/GroupListRow.c build/HistoryManager.c build/IMessageListRow.c build/InlineFileMessageListRow.c build/InlineImageMessageListRow.c build/MainWindow.c build/MessageListRow.c build/Notification.c build/PlainLabel.c build/ProfileChooser.c build/QuoteLabel.c build/QuoteMessageListRow.c build/Ricin.c build/Settings.c build/SettingsView.c build/StatusMessageListRow.c build/SystemMessageListRow.c build/ThemeManager.c build/ToxIdenticons.c build/Util.c build/WelcomeView.c build/Wrapper.c
Compilation failed: 7 error(s), 36 warning(s)

/home/liam/Ricin/src/ChatView.vala:46.14-46.32: warning: Ricin.ChatView.notify hides inherited field `GLib.Object.notify'. Use the `new' keyword if hiding was intentional
  [GtkChild] Gtk.Revealer notify;
             ^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/Wrapper.vala:656.9-656.53: warning: unhandled error `GLib.Error'
        tox.avatar.save_to_buffer (out pixels, "png");
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/Wrapper.vala:116.21-116.51: warning: unhandled error `GLib.Error'
      this.avatar = new Gdk.Pixbuf.from_file (path);
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/Wrapper.vala:404.19-404.96: warning: unhandled error `GLib.Error'
      var bytes = resources_lookup_data ("/chat/tox/ricin/nodes.json", ResourceLookupFlags.NONE);
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/Wrapper.vala:405.11-405.72: warning: unhandled error `GLib.Error'
      if (json.load_from_data ((string) bytes.get_data (), bytes.length)) {
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/Wrapper.vala:162.11-162.62: warning: unhandled error `GLib.FileError'
          FileUtils.get_data (profile, out opts.savedata_data);
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/Wrapper.vala:165.11-165.73: warning: unhandled error `GLib.Error'
          File.new_for_path (profile).create (FileCreateFlags.NONE, null);
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/Wrapper.vala:371.5-371.26: warning: Tox.Tox.disconnect hides inherited method `GLib.Object.disconnect'. Use the `new' keyword if hiding was intentional
    public void disconnect () {
    ^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/Wrapper.vala:526.9-526.47: warning: unhandled error `GLib.FileError'
        FileUtils.set_data (this.profile, data);
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/Wrapper.vala:675.18-675.76: warning: unhandled error `GLib.Error'
      var info = file.query_info ("standard::size", FileQueryInfoFlags.NONE);
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/Wrapper.vala:678.7-678.41: warning: unhandled error `GLib.FileError'
      FileUtils.get_data (path, out data);
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/ProfileChooser.vala:21.27-21.72: warning: unhandled error `GLib.Error'
    Gdk.Pixbuf app_icon = new Gdk.Pixbuf.from_resource (Ricin.ICON_PATH);
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/ProfileChooser.vala:34.7-34.27: warning: unhandled error `GLib.Error'
      dir.make_directory ();
      ^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/ProfileChooser.vala:36.24-36.78: warning: unhandled error `GLib.Error'
      var enumerator = dir.enumerate_children (FileAttribute.STANDARD_NAME, 0);
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/ProfileChooser.vala:39.22-39.44: warning: unhandled error `GLib.Error'
      while ((info = enumerator.next_file ()) != null) {
                     ^^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/Ricin.vala:55.7-55.63: warning: unhandled error `GLib.Error'
      config_sample.copy (config_file, FileCopyFlags.OVERWRITE);
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/MainWindow.vala:50.31-50.65: error: GLib.ListStore.new is not available in gio-2.0 2.40.2. Use gio-2.0 >= 2.44
  private ListStore friends = new ListStore (typeof (Tox.Friend));
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/FriendListRow.vala:54.20-54.81: warning: unhandled error `GLib.Error'
      var pixbuf = new Gdk.Pixbuf.from_file_at_scale (avatar_path, 48, 48, false);
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/MainWindow.vala:277.29-277.47: error: GLib.ListModel.get_n_items is not available in gio-2.0 2.40.2. Use gio-2.0 >= 2.44
          friend.position = friends.get_n_items ();
                            ^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/MainWindow.vala:279.11-279.24: error: GLib.ListStore.append is not available in gio-2.0 2.40.2. Use gio-2.0 >= 2.44
          friends.append (friend);
          ^^^^^^^^^^^^^^
/home/liam/Ricin/src/MainWindow.vala:272.22-272.51: warning: unhandled error `Tox.ErrFriendAdd'
        var friend = tox.accept_friend_request (id);
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/MainWindow.vala:307.27-307.45: error: GLib.ListModel.get_n_items is not available in gio-2.0 2.40.2. Use gio-2.0 >= 2.44
        friend.position = friends.get_n_items ();
                          ^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/MainWindow.vala:309.9-309.22: error: GLib.ListStore.append is not available in gio-2.0 2.40.2. Use gio-2.0 >= 2.44
        friends.append (friend);
        ^^^^^^^^^^^^^^
/home/liam/Ricin/src/MainWindow.vala:385.25-385.43: error: GLib.ListModel.get_n_items is not available in gio-2.0 2.40.2. Use gio-2.0 >= 2.44
      friend.position = friends.get_n_items ();
                        ^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/MainWindow.vala:389.7-389.25: error: GLib.ListStore.append is not available in gio-2.0 2.40.2. Use gio-2.0 >= 2.44
      this.friends.append (friend);
      ^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/MainWindow.vala:74.27-74.72: warning: unhandled error `GLib.Error'
    Gdk.Pixbuf app_icon = new Gdk.Pixbuf.from_resource (Ricin.ICON_PATH);
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/MainWindow.vala:140.20-140.74: warning: unhandled error `GLib.Error'
      var pixbuf = new Gdk.Pixbuf.from_file_at_scale (path, 48, 48, false);
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/MainWindow.vala:441.23-441.38: warning: unhandled error `Tox.ErrFriendDelete'
        bool result = friend.delete ();
                      ^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/MainWindow.vala:761.34-761.101: warning: unhandled error `GLib.Error'
      this.avatar_image.pixbuf = new Gdk.Pixbuf.from_file_at_scale (avatar.get_path (), 46, 46, true);
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/ChatView.vala:211.24-211.84: warning: unhandled error `GLib.Error'
          var pixbuf = new Gdk.Pixbuf.from_file_at_scale (avatar_path, 46, 46, true);
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/ChatView.vala:245.24-245.84: warning: unhandled error `GLib.Error'
          var pixbuf = new Gdk.Pixbuf.from_file_at_scale (avatar_path, 46, 46, true);
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/InlineFileMessageListRow.vala:73.9-73.76: warning: unhandled error `GLib.FileError'
        FileUtils.set_data (file_destination.get_path (), bytes.get_data ());
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/InlineFileMessageListRow.vala:77.9-77.76: warning: unhandled error `GLib.FileError'
        FileUtils.set_data (file_destination.get_path (), bytes.get_data ());
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/ChatView.vala:112.20-112.82: warning: unhandled error `GLib.Error'
      var pixbuf = new Gdk.Pixbuf.from_file_at_scale (_avatar_path, 48, 48, false);
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/InlineImageMessageListRow.vala:65.9-65.76: warning: unhandled error `GLib.FileError'
        FileUtils.set_data (file_destination.get_path (), bytes.get_data ());
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/InlineImageMessageListRow.vala:69.9-69.76: warning: unhandled error `GLib.FileError'
        FileUtils.set_data (file_destination.get_path (), bytes.get_data ());
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/InlineImageMessageListRow.vala:75.22-75.95: warning: unhandled error `GLib.Error'
        var pixbuf = new Gdk.Pixbuf.from_file_at_scale (this.image.get_path (), 400, 250, true);
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/InlineImageMessageListRow.vala:77.25-77.64: warning: unhandled error `GLib.Error'
        FileInfo info = this.image.query_info ("standard::*", 0);
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/InlineImageMessageListRow.vala:41.20-41.93: warning: unhandled error `GLib.Error'
      var pixbuf = new Gdk.Pixbuf.from_file_at_scale (this.image.get_path (), 400, 250, true);
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/InlineImageMessageListRow.vala:43.23-43.62: warning: unhandled error `GLib.Error'
      FileInfo info = this.image.query_info ("standard::*", 0);
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/InlineImageMessageListRow.vala:98.5-98.59: warning: unhandled error `GLib.Error'
    this.image.copy (image_destination, FileCopyFlags.NONE);
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/InlineImageMessageListRow.vala:101.7-101.26: warning: unhandled error `GLib.Error'
      this.image.delete ();
      ^^^^^^^^^^^^^^^^^^^^
/home/liam/Ricin/src/ChatView.vala:492.23-492.56: warning: unhandled error `GLib.Error'
      FileInfo info = file.query_info ("standard::*", 0);
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Waf: Leaving directory `/home/liam/Ricin/build'
Build failed
 -> task in 'Ricin' failed (exit status 1): 
    {task 47001634267792: valac ricin.gresource.xml,ChatView.vala,EditableLabel.vala,FriendListRow.vala,GroupListRow.vala,HistoryManager.vala,IMessageListRow.vala,InlineFileMessageListRow.vala,InlineImageMessageListRow.vala,MainWindow.vala,MessageListRow.vala,Notification.vala,PlainLabel.vala,ProfileChooser.vala,QuoteLabel.vala,QuoteMessageListRow.vala,Ricin.vala,Settings.vala,SettingsView.vala,StatusMessageListRow.vala,SystemMessageListRow.vala,ThemeManager.vala,ToxIdenticons.vala,Util.vala,WelcomeView.vala,Wrapper.vala -> ChatView.c,EditableLabel.c,FriendListRow.c,GroupListRow.c,HistoryManager.c,IMessageListRow.c,InlineFileMessageListRow.c,InlineImageMessageListRow.c,MainWindow.c,MessageListRow.c,Notification.c,PlainLabel.c,ProfileChooser.c,QuoteLabel.c,QuoteMessageListRow.c,Ricin.c,Settings.c,SettingsView.c,StatusMessageListRow.c,SystemMessageListRow.c,ThemeManager.c,ToxIdenticons.c,Util.c,WelcomeView.c,Wrapper.c}
''
make: *** [build] Error 1
```

This is with the latest Ricin. I tried to set it up this morning and have been trying to get it to work ever since. If more information is needed please let me know. I believe this is an issue specific to me...

**Comment by SkyzohKey** [comment]
Reading the backtrace I saw that : `Use gio-2.0 >= 2.44`. You'd probably need to update you gio version package. Thanks for reporting this, i'll add versions check to the `wscript` file ;)

**Comment by quorth0n** [comment]
Alright, once I figure out how to update gio-2.0 I'll report back

(Nice profile pic btw)

**Comment by SkyzohKey** [comment]
Assuming you fixed the issue by updating `glib-2.0` to version `>= 2.44`. Thus closing. Feel free to reopen if updating `glib` version doesn't fix the issue.
